huaweicloud/services/dc/resource_huaweicloud_dc_virtual_interface.go

```diff
@@ -0,0 +1,442 @@
+	}
+	return nil
+}
+
+func closeVirtualInterfaceNetworkDetection(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	var (
+		interfaceId = d.Id()
+		opts        = interfaces.UpdateOpts{}
+	)
+
+	// At the same time, only one of BFD and NQA is enabled.
+	if d.HasChange("enable_bfd") && !d.Get("enable_bfd").(bool) {
+		opts.EnableBfd = utils.Bool(false)
+	} else if d.HasChange("enable_nqa") && !d.Get("enable_nqa").(bool) {
+		opts.EnableNqa = utils.Bool(false)
+	}
+	if reflect.DeepEqual(opts, interfaces.UpdateOpts{}) {
+		return nil
+	}
+
+	_, err := interfaces.Update(client, interfaceId, opts)
+	if err != nil {
+		return fmt.Errorf("error closing network detection of the virtual interface (%s): %s", interfaceId, err)
+	}
+	return nil
+}
+
+func openVirtualInterfaceNetworkDetection(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	var (
+		interfaceId     = d.Id()
+		detectionOpened = false
+		opts            = interfaces.UpdateOpts{}
+	)
+
+	if d.HasChange("enable_bfd") && d.Get("enable_bfd").(bool) {
+		detectionOpened = true
+		opts.EnableBfd = utils.Bool(true)
+	}
+	if d.HasChange("enable_nqa") && d.Get("enable_nqa").(bool) {
+		// The enable requests of BFD and NQA cannot be sent at the same time.
+		if detectionOpened {
+			return fmt.Errorf("BFD and NQA cannot be enabled at the same time")
+		}
+		opts.EnableNqa = utils.Bool(true)
+	}
+	if reflect.DeepEqual(opts, interfaces.UpdateOpts{}) {
+		return nil
+	}
+
+	_, err := interfaces.Update(client, interfaceId, opts)
+	if err != nil {
+		return fmt.Errorf("error opening network detection of the virtual interface (%s): %s", interfaceId, err)
+	}
+	return nil
+}
+
+func resourceVirtualInterfaceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.DcV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating DC v3 client: %s", err)
+	}
+
+	if d.HasChanges("name", "description", "bandwidth", "remote_ep_group") {
+		var (
+			interfaceId = d.Id()
+
+			opts = interfaces.UpdateOpts{
+				Name:          d.Get("name").(string),
+				Description:   utils.String(d.Get("description").(string)),
+				Bandwidth:     d.Get("bandwidth").(int),
+				RemoteEpGroup: utils.ExpandToStringList(d.Get("remote_ep_group").([]interface{})),
+			}
+		)
+
+		_, err := interfaces.Update(client, interfaceId, opts)
+		if err != nil {
+			return diag.Errorf("error closing network detection of the virtual interface (%s): %s", interfaceId, err)
+		}
+	}
+	if d.HasChanges("enable_bfd", "enable_nqa") {
+		// BFD and NQA cannot be enabled at the same time.
+		// When BFD (NQA) is enabled and NQA (BFD) is disabled, we need to disable BFD (NQA) first, and then enable NQA (BFD).
+		// If the disable and enable requests are sent at the same time, an error will be reported.
+		if err = closeVirtualInterfaceNetworkDetection(client, d); err != nil {
```

Comment:
you can merge the two functions with
```
opts.EnableNqa = utils.Bool(d.Get("enable_bfd").(bool) == true)
```

Reply:
The create request and update request of the NQA and BFD cannot be sent at the same time.